File: problems/boj/9506.rs
// temp
fn main() {
    loop {
        let t: i32 = {
            let mut s = String::new();
            std::io::stdin().read_line(&mut s).unwrap();
            s.trim().parse::<i32>().unwrap()
        };
        if t == -1 {
            break;
        }
        let mut v: Vec<i32> = Vec::new();
        let half: i32 = t / 2 + 1;
        for i in 1..half {
            if t % i == 0 {
                v.push(i);
            }
        }
        let sum: i32 = v.iter().sum();
        if sum > t || sum != t {
            println!("{} is NOT perfect.", t);
        } else {
            let joined: Vec<String> = v.iter().map(|&s| s.to_string()).collect();
            println!("{} = {:?}", t, joined.join(" + "));
        }
    }
}

File: problems/boj/9506.py
while True:
  a = int(input())
  if a == -1:
    break
  n = []
  for i in range(1, int(a / 2) + 1):
    if a % i == 0:
      n.append(i)
  if sum(n) == a:
    print('{0} = {1}'.format(a, ' + '.join(list(map(str, n)))))
  else:
    print('{0} is NOT perfect.'.format(a))